packages/generators/remove-generator.ts

```diff
@@ -0,0 +1,136 @@
+import * as fs from "fs";
+import * as path from "path";
+import Generator = require("yeoman-generator");
+
+import * as PROP_TYPES from "@webpack-cli/utils/prop-types";
+import { List } from "@webpack-cli/webpack-scaffold";
+import { IWebpackOptions } from "./types";
+
+/**
+ *
+ * Generator for removing properties
+ * @class	RemoveGenerator
+ * @extends	Generator
+ * @returns	{Void} After execution, transforms are triggered
+ *
+ */
+
+export default class RemoveGenerator extends Generator {
+	private configuration: {
+		config: {
+			configName?: string,
+			topScope?: string[],
+			webpackOptions?: IWebpackOptions,
+		},
+	};
+	private webpackOptions: IWebpackOptions | string;
+
+	constructor(args, opts) {
+		super(args, opts);
+		this.configuration = {
+			config: {
+				webpackOptions: {},
+			},
+		};
+
+		let configPath: string = path.resolve(process.cwd(), "webpack.config.js");
+		const webpackConfigExists: boolean = fs.existsSync(configPath);
+		if (!webpackConfigExists) {
+			configPath = null;
+			// end the generator stating webpack config not found or to specify the config
+		}
+		this.webpackOptions = fs.readFileSync(configPath, "utf8");
```

Comment:
I think you'll need to do a JSON parse too, this will be a utf8 encoded string

Reply:
Yeah. This will return a string. Doing `JSON.parse()` on that string would result in error because the string contents aren't in JSON convertable format as it would be `module.exports = { ... }`. Even if we somehow read it in some JS variable, we would have to read object at `module.exports`.

I think the best way is to do `require(configPath)`. Same as we were doing earlier. It would directly read the webpack config object present at `module.export = { ... }`

Reply:
Yes you are right on the part on requiring the correct module, but when we do a require it will try to resolve dependencies we might not have in the repo and access variables that we might not need to bother with. It will result in a lot of errors for us and that's not good. 

Try to implement this using a JSON read instead. It is fully possible. To access the variables you'll need to do something like module['entry'] etc

Reply:
> it will try to resolve dependencies we might not have in the repo.

`require` doesn't always / only resolve npm dependencies. It's also a [standard convention for importing modules](https://nodejs.org/api/modules.html#modules_modules). If it's a file path like in this case, it won't try to resolve dependencies but simply import the module.exports object.

> It will result in a lot of errors for us and that's not good.

We were already using this convention in the existing codebase (master).

https://github.com/webpack/webpack-cli/blob/617af0bddb5140c07b1a0079b377311b10dd027d/packages/utils/modify-config-helper.js#L71

https://github.com/webpack/webpack-cli/blob/617af0bddb5140c07b1a0079b377311b10dd027d/packages/generators/add-generator.js#L326

Reply:
Can we have some third opinion @ematipico ?

Reply:
In that case, it should be changed. @dhruvdutt either way you do this, make sure not to import a file directly, as it would resolve the dependencies. I assume the closest thing you'll get to a error-free case is a stringified and parsed json